packages/features/bookings/Booker/utils/event.ts

```diff
@@ -14,15 +14,35 @@ import { useBookerStore } from "../store";
  * Using this hook means you only need to use one hook, instead
  * of combining multiple conditional hooks.
  */
-export const useEvent = () => {
+export const useEvent = (initialValues?: {
+  username: string;
+  eventSlug: string;
+  isTeamEvent: boolean;
+  org: string | null | undefined;
+}) => {
   const [username, eventSlug] = useBookerStore((state) => [state.username, state.eventSlug], shallow);
   const isTeamEvent = useBookerStore((state) => state.isTeamEvent);
   const org = useBookerStore((state) => state.org);
+  const params = {
+    username: username ?? initialValues?.username ?? "",
+    eventSlug: eventSlug ?? initialValues?.eventSlug ?? "",
+    isTeamEvent: isTeamEvent ?? initialValues?.isTeamEvent ?? false,
+    org: org ?? initialValues?.org ?? null,
+  };
```

Comment:
Store doesn't provide these values on first render, Booker has them so, use the provided values as the initial values and later let the store handle any changes in them.